python/packages/autogen-agentchat/src/autogen_agentchat/messages.py

```diff
@@ -4,18 +4,30 @@
 class and includes specific fields relevant to the type of message being sent.
 """
 
-from abc import ABC
-from typing import Dict, List, Literal
+from abc import ABC, abstractmethod
+from typing import Any, Dict, Generic, List, Literal, Mapping, TypeVar
 
 from autogen_core import FunctionCall, Image
 from autogen_core.memory import MemoryContent
-from autogen_core.models import FunctionExecutionResult, LLMMessage, RequestUsage
-from pydantic import BaseModel, ConfigDict, Field
-from typing_extensions import Annotated
+from autogen_core.models import FunctionExecutionResult, LLMMessage, RequestUsage, UserMessage
+from pydantic import BaseModel, ConfigDict, Field, computed_field
+from typing_extensions import Annotated, Self
 
 
 class BaseMessage(BaseModel, ABC):
-    """Base class for all message types."""
+    """Base class for all message types in AgentChat.
+
+    .. warning::
+
+        If you want to create a new message type, do not inherit from this class.
+        Instead, inherit from :class:`BaseChatMessage` or :class:`BaseAgentEvent`
+        to clarify the purpose of the message type.
+
+    """
+
+    content: Any
+    """The content of the message. The type is expected to be specified by the
+    subclass."""
```

Comment:
Can we leave this field to the subclass to define rather than it being Any here?

Reply:
True

Reply:
Actually I think we should keep the `content` field, because if we remove it, it will be a breaking change.